src/load_inputs/load_generators_data.jl

```diff
@@ -501,4 +496,100 @@ function load_vre_stor_data!(inputs_gen::Dict, setup::Dict, path::AbstractString
 		inputs_gen["dfVRE_STOR"] = DataFrame()
 	end
 	summarize_errors(error_strings)
-end
\ No newline at end of file
+end
+
+
+function process_piecewisefuelusage!(inputs::Dict, scale_factor)
+	gen_in = inputs["dfGen"]
+	inputs["PWFU_Num_Segments"] = 0
+	inputs["THERM_COMMIT_PWFU"] = Int64[]
+
+	if any(occursin.(Ref("PWFU_"), names(gen_in)))
+		heat_rate_mat = extract_matrix_from_dataframe(gen_in, "PWFU_Heat_Rate_MMBTU_per_MWh")
+		load_point_mat = extract_matrix_from_dataframe(gen_in, "PWFU_Load_Point_MW")
+		
+		# it's possible to construct piecewise fuel consumption with n of heat rate and n-1 of load point. 
+		# if a user feed n of heat rate and more than n of load point, throw a error message, and then use 
+		# n of heat rate and n-1 load point to construct the piecewise fuel usage fuction  
+		if size(heat_rate_mat)[2] < size(load_point_mat)[2]
+		 	@error """ The numbers of heatrate data are less than load points, we found $(size(heat_rate_mat)[2]) of heat rate,
+		 	and $(size(load_point_mat)[2]) of load points. We will just use the $(size(heat_rate_mat)[2]) of heat rate, and $(size(heat_rate_mat)[2]-1)
+			load points to create piecewise fuel usage
+		 	"""
+		end
+
+        # check if values for piecewise fuel consumption make sense. Negative heat rate or load point are not allowed
+        if any(heat_rate_mat .< 0) .| any(load_point_mat .< 0)
+			@error """ Neither heat rate nor load point can be negative
+			"""
+			error("Invalid inputs detected for piecewise fuel usage")
+		end
+		# for non-zero values, heat rates and load point should follow an increasing trend 
+		if any([any(diff(filter(x->x!=0, row)) .< 0) for row in eachrow(heat_rate_mat)]) .| any([any(diff(filter(x->x!=0, row)) .< 0) for row in eachrow(load_point_mat)])
```

Comment:
Can this conditional be broken into multiple lines? This is very long.

Reply:
`filter(x->x!=0, row)` 

I learned recently that `x->x!=0` can be written simply as `!=(0)`. Comparison operators now have functional forms.